### ISSUE TITLE
Updating fields

### ISSUE THREAD

**Issue by MatthewClarkMay** [tortoise/tortoise-orm#424]
I have code that polls an API for objects, every time those objects are polled I want to insert the object if it doesn't exist (custom pk) and update all fields if it does exist. Right now I am doing something like this...
```
fields = list(data.__dict__.keys())
unwanted = ['_partial', '_saved_in_db', '_custom_generated_pk', 'id']
cleaned = [field for field in fields if field not in unwanted]
await data.save(update_fields=cleaned)
```

It is throwing this exception...
```
duplicate key value violates unique constraint "data_pkey"
DETAIL:  Key (id)=(9953b156-5574-4086-a69f-ab4303659558) already exists.
```

Shouldn't it be updating values that already exist rather than throwing exceptions?

**Comment by MatthewClarkMay** [comment]
I should also add that I only want to update all fields if any of them have changed.

**Comment by MatthewClarkMay** [comment]
wrong place for this question, moving to gitter

**Comment by grigi** [comment]
Nah, this question needed a code change to become more elegant.

Current solution:
1. `update_fields` implies an update, so it will try harder to do an update.
2. `force_update=True` will force an update.
3. `force_create=True` will force a create.

The reason for the current behaviour is that we assume that if you create an object via the constructor it is new. That is being challenged, but we don't always know if we should do an insert or update. These changes should make it a bit easier for the user to enforce a certain behaviour.

You can test the behaviour by doing a `pip install https://github.com/tortoise/tortoise-orm/archive/develop.zip`

**Comment by grigi** [comment]
Released 0.16.13 with changes :smile: 
Hope this helps
